packages/app/src/api/visualization.js

```diff
@@ -1,17 +1,31 @@
 import { getFieldsStringByType } from '../modules/fields'
 
-const visualizationQuery = {
-    visualization: {
-        resource: 'visualizations',
-        id: ({ id }) => id,
-        params: {
-            fields: getFieldsStringByType('visualization'),
+export const apiFetchVisualization = (dataEngine, id) => {
+    const query = {
+        visualization: {
+            resource: 'visualizations',
```

Comment:
About this change, see https://github.com/dhis2/data-visualizer-app/pull/537#discussion_r364139186.

Reply:
The _id_ part is still there. But why is it preferable to keep the query as a _const_ instead of in the function? @amcgee

Reply:
It’s a semantic rule we put in place, so that queries can be shared or stored centrally.  It will also help us with caching in the future.  This is modeled after the Apollo / GraphQL approach.  Queries should be static and customized with variables - you will receive a console warning if they are not stable.

Reply:
Additionally, and most importantly, it enforces a 1-to-1 mapping of query to hook. Since the data is only fetched when you first mount the component, having a fully dynamic query doesn’t make sense - changing the query will either do nothing or lead to very strange behavior on a refetch, since the rendering code will likely assume a particular query response structure which would change if you for some reason decided to change the query resource to something else.

Reply:
Looking at this instance specifically it’s actually not a huge issue, though, since this is using the imperative rather than the declarative accessor.

Reply:
I reverted the change for `apiFetchVisualization`. For now `apiFetchVisualizations` is still using a self contained query, but I'd be happy to change it to follow the other example if someone can give me some guidance on how to get the variables (`filter, fields`) to work?

Reply:
Can't you pass all you need in `variables`?

Reply:
Like this? Can't get it to work...

`const visualizationsQuery = {
    visualization: {
        resource: 'visualizations',
        params: {
            filter: ({ filter }) => filter,
            fields: ({ fields }) => fields,
        },
    },
}

export const apiFetchVisualizations = (dataEngine, filter, fields) => {
    return dataEngine.query(visualizationsQuery, {
        variables: { filter, fields },
    })
}`

![image](https://user-images.githubusercontent.com/12590483/73161462-921f5d80-4111-11ea-9d28-8abc367bf9b5.png)

Reply:
Fixed! Could you guys take a quick look and re-review please?